lib/evaluation/limitDisclosureEvaluationHandler.ts

```diff
@@ -0,0 +1,158 @@
+  constructor(client: EvaluationClient) {
+    super(client);
+  }
+
+  public getName(): string {
+    return 'LimitDisclosureEvaluation';
+  }
+
+  //TODO: what is the necessary field? "@context", "credentialSchema", "credentialSubject", "type"
+  static mandatoryFields: string[] = ['@context', 'id', 'credentialSchema', 'credentialSubject', 'type'];
+
+  public handle(pd: PresentationDefinition, p: unknown): void {
+    for (let i = 0; i < pd.input_descriptors.length; i++) {
+      const constraints: Constraints = pd.input_descriptors[i].constraints;
+      //TODO: write the impl for "limitDisclosureShouldBeEnforced" as well. Should it generate WARNING?
+      if (constraints && constraints.limit_disclosure && constraints.limit_disclosure === Optionality.Required) {
+        this.limitDisclosureShouldBeEnforced(p, constraints.fields, i, pd.input_descriptors[i].id);
+      }
+    }
+  }
+
+  private limitDisclosureShouldBeEnforced(
+    verifiablePresentation: any,
+    fields: Field[],
+    idIdx: number,
+    inputDescriptorId: string
+  ): void {
+    for (let i = 0; i < verifiablePresentation.verifiableCredential.length; i++) {
+      const verifiableCredentialToSend = {};
+      let keys = Object.keys(verifiablePresentation.verifiableCredential[i]);
+      keys = this.copyMandatoryFieldsAndDeletePredefinedKeys(
+        verifiablePresentation.verifiableCredential[i],
+        verifiableCredentialToSend,
+        keys
+      );
+      this.determineNecessaryPaths(
+        verifiablePresentation.verifiableCredential[i],
+        verifiableCredentialToSend,
+        keys,
+        fields,
+        idIdx,
+        i
+      );
+      if (
+        this.verifiablePresentation.presentationSubmission &&
+        this.verifiablePresentation.presentationSubmission.descriptor_map
+      ) {
+        this.copyModifiedVerifiableCredentialToExisting(verifiableCredentialToSend, inputDescriptorId);
+      }
+    }
+  }
+
+  private copyMandatoryFieldsAndDeletePredefinedKeys(
+    verifiableCredential: unknown,
+    verifiableCredentialToSend: unknown,
+    keys: string[]
+  ): string[] {
+    for (let i = 0; i < LimitDisclosureEvaluationHandler.mandatoryFields.length; i++) {
+      verifiableCredentialToSend[LimitDisclosureEvaluationHandler.mandatoryFields[i]] =
+        verifiableCredential[LimitDisclosureEvaluationHandler.mandatoryFields[i]];
+      const index = keys.indexOf(LimitDisclosureEvaluationHandler.mandatoryFields[i]);
+      if (index > -1) {
+        keys.splice(index, 1);
+      }
+    }
+    return keys;
+  }
+
+  private determineNecessaryPaths(
+    vc: unknown,
+    vcToSend: unknown,
+    _keys: string[],
+    fields: Field[],
+    idIdx: number,
+    vcIdx: number
+  ) {
+    for (let i = 0; i < fields.length; i++) {
+      const field: Field = fields[i];
+      const result = JsonPathUtils.extractInputField(vc, field.path);
+      if (result.length > 0) {
+        //TODO: do we need to consider other paths here?
```

Comment:
Is the comment still necessary? Can we remove it?

Reply:
removed

Reply:
removed